byu-browser-oauth_test.mjs

```diff
@@ -148,12 +174,50 @@ describe('byu-browser-oauth', function () {
         });
     });
     describe('login()', function() {
-        it('exists');
+        it('dispatches login event to provider', async function () {
+            const prov = new FakeProvider();
+            afterEach(() => prov.disconnect());
+
+            const result = await authn.login();
+            expect(result.state).to.eql(authn.STATE_AUTHENTICATED);
+        });
     });
     describe('logout()', function() {
-        it('exists');
+        it('dispatches logout event to provider', async function () {
+            const prov = new FakeProvider({
+                state: authn.STATE_AUTHENTICATED,
+                user: fakeUser,
+                token: fakeToken,
+            });
+            afterEach(() => prov.disconnect());
+
+            expect(await authn.logout()).to.eql({ state: authn.STATE_UNAUTHENTICATED, user: null, token: null })
+        });
     });
     describe('refresh()', function() {
-        it('exists');
+        it('dispatches refresh event to provider', async function () {
+            const prov = new FakeProvider();
+            afterEach(() => prov.disconnect());
+
+            const result = await authn.refresh('background');
+            expect(result.state).to.eql(authn.STATE_AUTHENTICATED);
+            // NOTE: The "mode" parameter for authn.refresh normally does
```

Comment:
Is there any reason we couldn't say 'Hey, you can have extra, provider-specific things on the result object' and have a 'mode' field?  I think that'd be a nice thing in the implicit grant stuff if we add things like background refreshes.

Reply:
Yeah, poor word choice there: background refreshes in implicit grant is the exact reason we added that extra parameter. I'll give that param a more generic name.